### DISCUSSION TITLE
port 8080

### DISCUSSION THREAD

**Discussion by ghost** [PrivateBin/PrivateBin#1097] [Q&A] [answered]
here is my vps setup

VPS Setup

VPS 'purchase/install' setup a TLD wordpress (non docker) site (working)

afterwards docker was installed (working) i installed portainer for management.

example of one container searxng installed run command (working)

sudo docker run --restart always -d -p 8080:8080 -v "${PWD}/searxng:/etc/searxng" -e "BASE_URL=http://localhost:8080/" -e "INSTANCE_NAME=searxng" searxng/searxng

i would like to install privatebin also

the run command direct from their docker guide is

docker run -d --restart="always" --read-only -p 8080:8080 -v $PWD/privatebin-data:/srv/data privatebin/nginx-fpm-alpine

i cant run the exposed port as 8080 since searxng is running. for example i tried adjusting the command from 8080:8080 to 7979:8080, it installs however when accessing say https://privatebin.tld.org i receive a 502 error from cloudflare. i am using cloudflare tunnels as FYI.

if i remove the searxng container and install privatebin as instructed 8080:8080, privatebin works throught https://subdomain.tld.org

any ideas?

**Comment by elrido** [comment, marked as answer]
You don't have to expose the container on the port 8080, thats just what it uses internally. The[ `--expose/-p` parameter](https://docs.docker.com/engine/reference/commandline/run/#publish) is a map of the external to the internal port for exposing container services. You can map any (not in use) external port to the internal one, so you could just expose PrivateBin on port 8081 and avoid the conflict: `-p 8081:8080` - in your external facing webserver/proxy (the one you likely exposed on ports 80 and 443) you can then proxy requests to the searxng and privatebin paths or sub-domains to their respective exposed ports, 8080 and 8081 in this example.

**Reply by ghost** [reply]
this worked! not only is privatebin working but so are all my other containers i was testing : ) great work mate!
